ui/app/src/components/VideoPlayer.tsx

```diff
@@ -16,17 +16,71 @@
  */
 
 import React from 'react';
+import videojs from 'video.js';
+import 'video.js/dist/video-js.css';
+import '@videojs/http-streaming';
 
-interface VideoPlayerProps {
-  // TODO: list props...
-}
+import { PlayerOptions } from '../models/PlayerOptions';
+
+const SourceTypes: { [key: string]: string } = {
+  'mp4': 'video/mp4',
+  'm3u8': 'application/x-mpegURL',
+  'mpd': 'application/dash+xml'
+};
+
+class VideoPlayer extends React.Component {
```

Comment:
Use/learn hooks, ditch the class :)

For `player` and `videoNode` -> useRef
For `videoJsOptions` -> useMemo

BTW, the optionality sign (`?`) on class properties doesn't make too much sense. That's for params and interfaces.

Ping me with questions.